src/TextField/TextField.jsx

```diff
@@ -370,6 +391,12 @@ const TextField = React.createClass({
     if (this.props.onKeyDown) this.props.onKeyDown(event);
   },
 
+  _handleInputKeyUp() {
+    this.setState({currentCharCount: this.getValue().length});
+    if (this.props.charCountMax >= this.getValue().length) this.setState({charCountError: false});
```

Comment:
I'm wondering if we supply the `value` to the `TextCharCount` component, it can perform this logic and we won't need to store any additional sate with the `TextField` component.